File: MyWatchlist/MyWatchlistAngular/src/app/models/series.ts
import {Genre} from "./genre";

export class Series {
  id: number;
  title: string;
  episodes: number;
  rating: number;
  plot: string;
  views: number;
  members: number;
  genres: Genre[];

  constructor(id:number, title:string, episodes:number,
              rating:number, plot:string, views:number,
              members:number, genres:Genre[]) {
    this.id = id;
    this.title = title;
    this.episodes = episodes;
    this.rating = rating;
    this.views = views;
    this.plot = plot;
    this.members = members;
    this.genres = genres;
  }
}

File: MyWatchlist/MyWatchlistAndroid/app/src/main/java/it/niko/mywatchlistandroid/model/Series.kt
package it.niko.mywatchlistandroid.model

import com.google.gson.annotations.SerializedName

data class Series(
    @SerializedName("title")
    val title: String,

    @SerializedName("episodes")
    val episodes: Int,

    @SerializedName("rating")
    val rating: Double,

    @SerializedName("views")
    val views: Int,

    @SerializedName("plot")
    val plot: String,

    @SerializedName("members")
    val members: Int,


    @SerializedName("genres")
    val genres: ArrayList<Genre>
)
